### ISSUE TITLE
When hash-pinning GitHub Actions, Dependabot updated an Action's version comment in one file but not another

### ISSUE THREAD

**Issue by pnacht** [dependabot/dependabot-core#7376]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package ecosystem

github-actions

### dependabot.yml content

```yml
version: 2
updates:
  - package-ecosystem: "github-actions" # See documentation for possible values
    directory: "/" # Location of package manifests
    schedule:
      interval: "weekly"
```

### What you expected to see, versus what you actually saw

We received a single PR updating actions/checkout in two files:

`scorecard.yml` is correctly updated, both the hash and the version comment:

```diff
    steps:
      - name: "Checkout code"
-        uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # v3.1.0
+        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
```

However, for `testing.yml` only the hash is updated:

```diff
    steps:
-      - uses: actions/checkout@83b7061638ee4956cf7545a6f7efe594e5ad0247 # v3.5.1
+      - uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.1
```

If relevant, the hash for v3.5.1 was manually obtained by going to the [actions/checkout Releases page](https://github.com/actions/checkout/releases/tag/v3.5.1).

I'd have expected `testing.yml` to also change the version comment to `v3.5.2`.

### Images of the diff or a link to the PR, issue, or logs

https://github.com/nebraska-dev/dependabot-debug/pull/5

![Screenshot 2023-05-29 at 18 41 44](https://github.com/dependabot/dependabot-core/assets/15221358/6b7efbbe-0eb6-43bc-9494-cd4161db7b9b)

### Smallest manifest that reproduces the issue

The repo linked above is pretty close to an MRE, but the following should (not) work as well:

`dependabot.yml`:

```yml
version: 2
updates:
  - package-ecosystem: "github-actions" # See documentation for possible values
    directory: "/" # Location of package manifests
    schedule:
      interval: "weekly"
```

`testing.yml`:

```yml
on:
  push:

jobs:
  build:
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@83b7061638ee4956cf7545a6f7efe594e5ad0247 # v3.5.1
```

**Comment by pnacht** [comment]
That particular case actually seems to be working as intended.

`v2` is a major version tag which is kept up-to-date with the latest release. So it used to point to [`v2.1.0`](https://github.com/docker/login-action/releases/tag/v2.1.0), but now it points to [`v2.2.0`](https://github.com/docker/login-action/releases/tag/v2.2.0).

If you change the comment to `# v2.2.0`, then you should expect to see it updated to `# v2.3.0` (or whatever the new version may be) on the next release.

**Comment by deivid-rodriguez** [comment]
This was also fixed by #8068 and #8098 :tada:. Let me know if you still see issues though!
